lib/pronto/gitlab.rb

```diff
@@ -27,6 +67,25 @@ def slug
       end
     end
 
+    def pull_id
+      pull ? pull.iid.to_i : env_pull_id
+    end
+
+    def pull
+      @pull ||= if env_pull_id
+                  pull_requests.find { |pr| pr.iid.to_i == env_pull_id }
+                elsif @repo.branch
+                  # TODO: Update for gitlab
+                  pull_requests.find do |pr|
+                    pr.source['branch']['name'] == @repo.branch
+                  end
+                end
+    end
+
+    def pull_requests
+      @pull_requests ||= client.merge_requests(slug)
```

Comment:
Maybe limit the MRs based on `state=open`? Branch names can be reused in which case `pull` could return a closed MR.

It is also possible to filter based on the `source_branch` and `id` which would speed up the request. https://docs.gitlab.com/ee/api/merge_requests.html